### PR TITLE
little fix for Identity resource

### PR DESCRIPTION
**Pull request by toha10**
This fix solves ''unhashable type dict' error in _get_user_tenants_roles_by_db method of Identity CF resource.

### CODE REVIEW

**Comment by miarmak** [comment]
Unfortunately, all functional tests failed with:
```
ERROR    [/home/vagrant/CloudFerry/cloudferrylib/scheduler/scheduler.py : lineno 58]  10-22 13:36:52  BaseTask|IdentityTransporter TASK FAILED
Traceback (most recent call last):
  File "/home/vagrant/CloudFerry/cloudferrylib/scheduler/scheduler.py", line 71, in process_chain
    self.run_task(task)
  File "/home/vagrant/CloudFerry/cloudferrylib/scheduler/scheduler.py", line 63, in run_task
    self.map_func_task[task](task)
  File "/home/vagrant/CloudFerry/cloudferrylib/scheduler/scheduler.py", line 94, in task_run
    task(namespace=self.namespace)
  File "/home/vagrant/CloudFerry/cloudferrylib/scheduler/task.py", line 97, in __call__
    result = self.run(**namespace.vars)
  File "/home/vagrant/CloudFerry/cloudferrylib/os/actions/identity_transporter.py", line 26, in run
    info = src_resource.read_info(**search_opts)
  File "/home/vagrant/CloudFerry/cloudferrylib/os/identity/keystone.py", line 188, in read_info
    self._get_user_tenants_roles(tenant_list, user_list)
  File "/home/vagrant/CloudFerry/cloudferrylib/os/identity/keystone.py", line 526, in _get_user_tenants_roles
    user_list)
  File "/home/vagrant/CloudFerry/cloudferrylib/os/identity/keystone.py", line 590, in _get_user_tenants_roles_by_db
    for r in roles_ids]
TypeError: string indices must be integers, not str
```
Please fix.

**Comment by miarmak** [comment]
Looks good, +1.

File: cloudferrylib/os/identity/keystone.py [hunk 1]
```diff
@@ -609,9 +609,14 @@ def _get_user_tenants_roles_by_db(self, tenant_list, user_list):
                 continue
 
             roles_ids = ast.literal_eval(roles_field)['roles']
+            if isinstance(roles_ids[0], dict):
```
Comment (roman-verchikov) [inline cloudferrylib/os/identity/keystone.py]: Verifying object type at runtime for these types of failures looks like a dirty hack. Also you try to retrieve first element of object (`roles_ids[0]`), which may result in `IndexError` for empty list.

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.identity.keystone
cloudferrylib/os/identity/keystone.py:195: [W0221(arguments-differ), KeystoneIdentity.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/identity/keystone.py:575: [W0612(unused-variable), KeystoneIdentity._get_roles_sql_request] Unused variable 'type_record'
cloudferrylib/os/identity/keystone.py:576: [W0612(unused-variable), KeystoneIdentity._get_roles_sql_request] Unused variable 'inher_tmp'
cloudferrylib/os/identity/keystone.py:622: [W0141(bad-builtin), KeystoneIdentity._get_user_tenants_roles_by_db] Used builtin function 'map'
cloudferrylib/os/identity/keystone.py:622: [W0110(deprecated-lambda), KeystoneIdentity._get_user_tenants_roles_by_db] map/filter on lambda could be replaced by comprehension

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/1413/

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.identity.keystone
cloudferrylib/os/identity/keystone.py:195: [W0221(arguments-differ), KeystoneIdentity.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/identity/keystone.py:575: [W0612(unused-variable), KeystoneIdentity._get_roles_sql_request] Unused variable 'type_record'
cloudferrylib/os/identity/keystone.py:576: [W0612(unused-variable), KeystoneIdentity._get_roles_sql_request] Unused variable 'inher_tmp'
cloudferrylib/os/identity/keystone.py:622: [W0141(bad-builtin), KeystoneIdentity._get_user_tenants_roles_by_db] Used builtin function 'map'
cloudferrylib/os/identity/keystone.py:622: [W0110(deprecated-lambda), KeystoneIdentity._get_user_tenants_roles_by_db] map/filter on lambda could be replaced by comprehension

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/1419/

File: cloudferrylib/os/identity/keystone.py [hunk 2]
```diff
@@ -617,9 +617,15 @@ def _get_user_tenants_roles_by_db(self, tenant_list, user_list):
                 continue
 
             roles_ids = ast.literal_eval(roles_field)['roles']
+            db_version = self.get_db_version()
+            if db_version in xrange(29, 39):
```
Comment (roman-verchikov) [inline cloudferrylib/os/identity/keystone.py]: Why not simply `29 <= db_version and db_version <= 39`? 2 simple comparisons is much faster than 10 comparisons in a loop. It is also as readable.

**Comment by MirantisWorkloadMobility** [comment]
Functional test success for Grizzly to Icehouse migration

**Comment by MirantisWorkloadMobility** [comment]
Functional test success for Icehouse to Juno migration

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.identity.keystone
cloudferrylib/os/identity/keystone.py:195: [W0221(arguments-differ), KeystoneIdentity.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/identity/keystone.py:575: [W0612(unused-variable), KeystoneIdentity._get_roles_sql_request] Unused variable 'type_record'
cloudferrylib/os/identity/keystone.py:576: [W0612(unused-variable), KeystoneIdentity._get_roles_sql_request] Unused variable 'inher_tmp'
cloudferrylib/os/identity/keystone.py:622: [W0141(bad-builtin), KeystoneIdentity._get_user_tenants_roles_by_db] Used builtin function 'map'
cloudferrylib/os/identity/keystone.py:622: [W0110(deprecated-lambda), KeystoneIdentity._get_user_tenants_roles_by_db] map/filter on lambda could be replaced by comprehension

**Comment by MirantisWorkloadMobility** [comment]
Functional test success for Grizzly to Juno migration

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/1420/

**Comment by MirantisWorkloadMobility** [comment]
Functional test failure for Grizzly to Juno migration
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-functional-tests-grizzly-to-juno/625/

**Comment by MirantisWorkloadMobility** [comment]
Functional test failure for Grizzly to Icehouse migration
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-functional-tests-grizzly-to-icehouse/1515/

**Comment by MirantisWorkloadMobility** [comment]
Functional test success for Icehouse to Juno migration

**Comment by Settis** [comment]
Functional test success for [Grizzly to Juno](http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-functional-tests-grizzly-to-juno/649)
Functional test success for [Grizzly to Icehouse](http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-functional-tests-grizzly-to-icehouse/1540)
+1

**Pull request merged**